.github/scripts/verify-macos-dylibs.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/python3
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright (C) 2021-2021  Patryk Obara <patryk.obara@gmail.com>
+
+# pylint: disable=invalid-name
+
+"""
+Verify if there are no problematic runtime dependencies for this macOS binary.
+"""
+
+import sys
+import subprocess
+
+
+if __name__ == '__main__':
+    result = subprocess.run(['otool', '-L', sys.argv[1]],
+                            stdout=subprocess.PIPE,
+                            check=True)
+    output = result.stdout.decode('utf-8').split('\n')
+
+    allowed_list = [
+        '/System/Library/Frameworks/AppKit.framework/',
+        '/System/Library/Frameworks/AudioToolbox.framework/',
+        '/System/Library/Frameworks/AudioUnit.framework/',
+        '/System/Library/Frameworks/Carbon.framework/',
+        '/System/Library/Frameworks/CoreAudio.framework/',
+        '/System/Library/Frameworks/CoreFoundation.framework/',
+        '/System/Library/Frameworks/CoreGraphics.framework/',
+        '/System/Library/Frameworks/CoreMIDI.framework/',
+        '/System/Library/Frameworks/CoreServices.framework/',
+        '/System/Library/Frameworks/CoreVideo.framework/',
+        '/System/Library/Frameworks/ForceFeedback.framework/',
+        '/System/Library/Frameworks/Foundation.framework/',
+        '/System/Library/Frameworks/GameController.framework/',
+        '/System/Library/Frameworks/IOKit.framework/',
+        '/System/Library/Frameworks/Metal.framework/',
+        '/System/Library/Frameworks/OpenGL.framework/',
+        '/usr/lib/',
+    ]
+
```

Comment:
I can't remember if `otool` shows it or not; but I remember seeing strong vs. weak linking.
From what I've read, weak-linking allows much easier backward compatibility because it has fewer restrictions.

So one option here would be to allow any weakly-linked libraries in `/System/Library/Frameworks`, leaving only the strongly-linked /System/Library/Frameworks libraries as things we need to check.

That might be a bit more tricky to do code-wise though, so perhaps this is easier.

Reply:
Hmm, I haven't seen anything about "strong/weak" linking in otool output.

So far I decided to allow everything in `/usr/lib` and everything that got linked from `/System/Library/Frameworks/*` (but put it one-by-one, so we know what is being linked) - this way we disallow everything from `/usr/local/ and `/opt/` - so if something will get pulled in by mistake after brew update, we'll see it due to failing release build.

Perhaps I am too pedantic here and allow *everything* from `/System/Library/Frameworks/`, without cherry-picking?

Reply:
Yes, /usr/local/ and /opt/ are the red flag we definitely need to catch.  

I think it will be interesting to catch new framework libs - and atleast understand why they're needed. In most cases there will be little we can do about it (unless we start building SDL ourselves and control it's configure settings). It will also help us correlate mac-breaking eventsnif users reports the binary is messing up (if there was a recent lib added).  

All that to say, I think it's good 👍